### PR TITLE
C++: Use local flow instead of GVN in `getAdditionalFlowIntoCallNodeTerm`

### PR DESCRIPTION
**Pull request by MathiasVP**
This fixes a performance issue we've observed on a project, where the function was too complex for `GVN` to figure out that a parameter was used in a switch statement. This meant that the performance-mitigation measure we implemented in https://github.com/github/codeql/pull/12236 didn't kick in.

Commit-by-commit recommended. All the meat is in the first commit, and the second commit just moved things around for caching purposes.

We're losing a bunch of DCA results, but OTOH we're fixing a performance regression on a high-valued project. I'll create an internal issue to discuss how to get back some of these results.

### CODE REVIEW

File: cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowPrivate.qll [hunk 1]
```diff
@@ -5,33 +5,59 @@ private import DataFlowDispatch
 private import DataFlowImplConsistency
 private import semmle.code.cpp.ir.internal.IRCppLanguage
 private import SsaInternals as Ssa
-private import DataFlowImplCommon
+private import DataFlowImplCommon as DataFlowImplCommon
 private import semmle.code.cpp.ir.ValueNumbering
```
Comment (jketema) [inline cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowPrivate.qll:9]: Do we still need this import?

File: cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowPrivate.qll [hunk 2]
```diff
@@ -897,6 +897,55 @@ IRBlock getBasicBlock(Node node) {
...
+    localStepsToSwitch(succ) and
+    localFlowStepWithSummaries(node, succ)
+  )
+}
+
+/**
+ * Holds if `node` is part of a path from a `ParameterNode` to an operand
+ * of a `SwitchInstruction`.
+ */
+private predicate localStepsFromParameter(Node node) {
```
Comment (jketema) [inline cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowPrivate.qll:951]: Nitpicking: `localStepsFromParameterToSwitch`?

Comment (MathiasVP) [reply cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowPrivate.qll]: Nitpicking accepted 😂: https://github.com/github/codeql/pull/12532/commits/08419b77af99dccb1d8760cbab1fa96a9d936de4

**Review by jketema — APPROVED** [review]
LGTM. For clarity: we agreed that the slight loss in DCA results is acceptable at this point and something we can revisit later.

**Pull request merged**